HPCC-18689 Avoid sharing method accessmap to avoid core

rpastrana opened the pull request:
- Removes sharing of method access map member var now created a smaller scope
- Merge service and method access flags and process in 1 pass

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@wangkx please review

HPCCSmoketest commented:
Automated Smoketest: :x:
Sha: 40347713fe87a2d2431ffdc91182802903bd96d7
Empty table
Build: failed
Number of Error(s): 120
Error: Cannot find module jshint/src/cli
[  0%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/CoreErrors.cpp.o
[100%] Building CXX object aws-cpp-sdk-sqs/CMakeFiles/aws-cpp-sdk-sqs.dir/source/SQSErrors.cpp.o
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:12918:8: error: \xe2\x80\x98Access\xe2\x80\x99 was not declared in this scope
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:12918:15: error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98level\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13034:1: error: duplicate label \xe2\x80\x98Error\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13034:8: error: \xe2\x80\x98Access\xe2\x80\x99 was not declared in this scope
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13034:15: error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98level\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13224:1: error: duplicate label \xe2\x80\x98Error\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13224:8: error: \xe2\x80\x98Access\xe2\x80\x99 was not declared in this scope
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13224:15: error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98level\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13414:1: error: duplicate label \xe2\x80\x98Error\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13414:8: error: \xe2\x80\x98Access\xe2\x80\x99 was not declared in this scope
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13414:15: error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98level\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13456:1: error: duplicate label \xe2\x80\x98Error\xe2\x80\x99
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoketest/PR-10644/build/generated/ws_smc.esp:13456:8: error: \xe2\x80\x98Access\xe2\x80\x99 was not declared in this scope
 Error: Access level must be declared in service definition: ESPservice [auth_feature(myAccessFeature:FULL]
/mnt/disk1/home/vamosax/smoket
 ... ( comment is too long, 128 lines, 15637 bytes, truncated) ... 
/ecl-roxie.dir/__/__/__/generated/ws_smc_esp.cpp.o] Error 1
make[1]: *** [ecl/ecl-bundle/CMakeFiles/ecl-bundle.dir/all] Error 2
make[1]: *** [ecl/eclcmd/roxie/CMakeFiles/ecl-roxie.dir/all] Error 2
make: *** [all] Error 2
Get ECLWatch build result


```

rpastrana committed:
HPCC-18689 Avoid sharing method accessmap to avoid core

- Removes sharing of method access map member var now created a smaller scope
- Merge service and method access flags and process in 1 pass

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: b0f871ea4b5b86ae592d87c51fd1255dcb5dd346
Empty table
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 95.384 seconds |


Install hpccsystems-platform-community_6.4.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |96 |96 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented:
@rpastrana looks fine.

ghalliday commented:
@richardkchapman. The change looks fine - a bit inefficient in the case where there is no checking (fairly common). 

For future, I'm not sure why it uses a hash table, when the code just iterates it to get a list.
